components/date-picker/standard-types.ts

```diff
@@ -17,9 +17,16 @@ export type CompatibleDate = Date | Date[];
 
 export type DisabledTimeFn = (current: Date | Date[], partial?: DisabledTimePartial) => DisabledTimeConfig | undefined;
 
+export interface NzPanelChangeType {
+  mode: NzDateMode | [startMode: NzDateMode, endMode: NzDateMode];
+  date: Date | [startDate: Date, endDate: Date];
+}
```

Comment:
```suggestion
export type NzPanelChangeType = {
  mode: NzDateMode;
  date: Date;
} | {
  mode: [startMode: NzDateMode, endMode: NzDateMode];
  date:  [start: Date, end: Date];
}
```

WDYT?

Reply:
yes, we could do that to have a more flexible schema